docs/docs/examples.md

```diff
@@ -0,0 +1,308 @@
+    contextString = "Signature1",
+    protectedHeader = ByteStringWrapper(protectedHeader),
+    externalAad = byteArrayOf(),
+    payload = payload,
+).serialize()
+```
+
+
+Now, everything is ready to be signed:
+
+```kotlin
+val signature = signer.sign(signatureInput).signature //TODO handle error
+
+val coseSigned = CoseSigned(
+    ByteStringWrapper(protectedHeader),
+    unprotectedHeader = null,
+    payload,
+    signature
+).serialize() // sadly, there's no cwt.io, but you can use cbor.me to explore the signed data
+```
+
+## Create and Parse a Custom-Tagged ASN.1 Structure
+
+!!! info  inline end
+    This example requires only the _Indispensable_ module.
+
+When you come across a certain pattern more than once, you only encode and decode it once and recycle this code.
+Still, it has to be done at least once. This example shows how to create a small, custom-tagged ASN.1 structure and shows how
+it can be parsed and validated.
+
+### Definitions
+
+Let's say you are using ASN.1 as your wire format and you want to report the status about an operation.
+The status message is an implicitly tagged ASN.1 structure with APPLICATION tag `1337` and sequence semantics.
+It contains the number of times the operation was run, and a timestamp, which can be either relative (in whole seconds since the last operation)
+or absolute (UTC Time). This relative/absolute flag uses the implicit APPLICATION tag `42` and the tuple of flag and time
+is encoded into an ANS.1 OCTET STRING. This allows for two possible encodings, as illustrated below:
+
+<table>
+<tr>
+<th>
+Absolute Time
+</th>
+<th>
+Relative Time
+</th>
+</tr>
+<tr>
+<td>
+
+```asn1
+Application 1337 (2 elem)
+  INTEGER 1
+  OCTET STRING (19 byte)
+    Application 42 (1 byte) 00
+    UTCTime 2024-09-30 18:11:59 UTC
+```
+
+</td>
+
+<td>
+
+```asn1
+Application 1337 (2 elem)
+  INTEGER 3
+  OCTET STRING (7 byte)
+    Application 42 (1 byte) FF
+    INTEGER 39
+```
+
+</td>
+</tr>
+</table>
+
+### Encoding
+
+We'll be assuming absolute time to keep things simple.
+Hence, the structure containing an absolute time can be created using the _Indispensable_ ASN.1 engine as follows:
+
+```kotlin
+val TAG_TIME_RELATIVE = 42uL withClass TagClass.APPLICATION
+
+Asn1.Sequence {
+    +Asn1.Int(1)
+    +OctetStringEncapsulating {
+        +(Bool(false) withImplicitTag TAG_TIME_RELATIVE)
+        +Asn1Time(Clock.System.now())
+    }
+} withImplicitTag (1337uL withClass TagClass.APPLICATION) 
+//                ↑ in reality this would be a constant ↑ 
+```
+
+The HEX-equivalent of this structure (which can be obtained by calling `.toDerHexString()`) is
+[7F8A391802010104135F2A0100170D3234303933303138313135395A](https://lapo.it/asn1js/#f4o5GAIBAQQTXyoBABcNMjQwOTMwMTgxMTU5Wg).
+
+### Parsing and Validating Tags
+
+Basic parsing is straight-forward: You have DER-encoded bytes, and feed them into `AsnElement.parse()`.
+Then you examine the first child to get the number of times the operation was carried out,
```

Comment:
this makes it sound like this applies to all ASN.1 structures; i don't think this is the case?